core/trashcan.js

```diff
@@ -428,38 +431,74 @@ Blockly.Trashcan.prototype.emptyContents = function() {
  * Position the trashcan.
  * It is positioned in the opposite corner to the corner the
  * categories/toolbox starts at.
- */
-Blockly.Trashcan.prototype.position = function() {
+ * @param {!Blockly.MetricsManager.ContainerRegion} viewMetrics The workspace
+ *     viewMetrics.
+ * @param {!Blockly.MetricsManager.AbsoluteMetrics} absoluteMetrics The absolute
+ *     metrics for the workspace.
+ * @param {!Blockly.MetricsManager.ToolboxMetrics} toolboxMetrics The toolbox
+ *     metrics for the workspace.
```

Comment:
Should we just pass in all the metrics? In case other positionables end up needing them?

Reply:
The downside to all the metrics is that contentMetrics is somewhat expensive and not something I would expect (or want...) a positionable to need.
That being said, I'm still open to passing a single metrics object. Going that route, I could also make a custom type that is passed so we have flexibility to adding what is included on the metrics object without needing to change the method signature.

Reply:
Ok, that sounds reasonable. And yes, I like the idea of passing a single metrics object.

Reply:
As long as it's fine with you. I'm going to look at making a single metrics object in next PR.